### DISCUSSION TITLE
Tumor-only-model

### DISCUSSION THREAD

**Discussion by RoniHaas** [uclahs-cds/pipeline-call-sSNV#88] [Q&A]
Here is the benchmarking result of the newly added Mutect2 option `--germline-resource`:

## Benchmarking
- Normal-paired-model : 
    - sample: A-full-p2
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/vcf_comparsion_Afull.txt

Option | Total SNPs | Unique SNPs and % | Shared SNPs and %
--|--|--|--
With `germline resource` |133,585|10; 0.0%| 133,575; 100.0%
 W. O.  `germline resource` |133,667|92; 0.1% | 133,575; 99.9%

    - sample: CPCG0196
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/CPCG_vcf_comparsion_paired.txt

Option | Total SNPs | Unique SNPs and % | Shared SNPs and %
--|--|--|--
With `germline resource` |34,747|30; 0.1%| 34717; 99%
 W. O.  `germline resource` |43,276|8559; 19.8% | 34717; 80.2%

- Tumor-only-model :
    - sample: anaplastic thyroid cancer sample - MDA10
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/vcf_comparsion.txt

Option | Total SNPs | %Unique SNPs  | %Shared SNPs
--|--|--|--
With `germline resource` |122,005 | 1488; 1.2% | 120517; 98.8%
 W. O.  `germline resource` |3,552,165 | 3431648; 96.6% | 120517; 3.4%

     - sample: CPCG0196
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/CPCG_vcf_comparsion_tumor_only.txt

Option | Total SNPs | Unique SNPs and % | Shared SNPs and %
--|--|--|--
With `germline resource` |149,053|2737; 1.8%| 146316; 98.2%
 W. O.  `germline resource` |189,592|3165672; 95.6% | 146316; 4.4%

- Multiple-tumor-model : 
    - sample:anaplastic thyroid cancer sample - ATCWGS21
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/vcf_comparsion_ATCWGS.txt

Option | Total SNPs | %Unique SNPs  | %Shared SNPs
--|--|--|--
With `germline resource` | 13,104|2; 0.0%| 13,102; 100%
 W. O.  `germline resource` | 13,561 |459; 3.4%|13,102; 96.6%

- Comparison between tumor-only and paired Mutect2
     - sample: CPCG0196
    - vcf comparison result: /hot/pipeline/development/slurm/call-sSNV/unreleased/maotian-add-germline-resource/CPCG_vcf_compare_paired_tumor_only.txt

Option | Total SNPs | %Unique SNPs  | %Shared SNPs
--|--|--|--
Paired Normal-Tumor| 34,747|14138;40.7%| 20,609; 59.3%
Tumor-only| 149,053 |128,444; 86.2%|20,609; 13.8%

_Originally posted by @maotian06 in https://github.com/uclahs-cds/pipeline-call-sSNV/discussions/80_

**Comment by RoniHaas** [comment]
> Originally posted by @maotian06 in #80

Hey @maotian06,
I am interested in running the call-sSNV in a Tumor-only-model but not so sure how to do that. Can you please help with some details?

**Comment by tnyamaguchi** [comment]
@RoniHaas this PR might help - https://github.com/uclahs-cds/pipeline-call-sSNV/tree/maotian-update-README

**Reply by RoniHaas** [reply]
> @RoniHaas Any details of your samples? I will close this issue and move it to the discussion.



> @RoniHaas this PR might help - https://github.com/uclahs-cds/pipeline-call-sSNV/tree/maotian-update-README

Thank you! All I need to do is to use this yaml template https://github.com/uclahs-cds/pipeline-call-sSNV/blob/maotian-update-README/input/example-test-tumor-only.yaml
@maotian06  @tyamaguchi-ucla
Should I change the config file?

**Reply by maotian06** [reply]
You can use the newest release here: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/release/4.0.0-rc.1/`
For `input.YAML`, you can copy `config/template.config` and this YAML to your own directory and change it: `/hot/software/pipeline/pipeline-call-sSNV/Nextflow/release/4.0.0-rc.1/input/example-test-tumor-only.yaml`

**Comment by maotian06** [comment]
@RoniHaas 
Any details of your samples?
I will close this issue and move it to the discussion.

**Reply by RoniHaas** [reply]
For call-gSNP, I ran in pairs as the N-T pair: T_radiosensitive-T_radioresistant. That means that my normal was actually a tumor sample. 
What I want now is to run the T_radiosensitive ( used as "normal") sample alone, to get an evaluation of the number of sSNVs. I don't have a matched normal sample. If it helps, I have 3 replicates for this T_radiosensitive sample.

**Reply by tnyamaguchi** [reply]
@RoniHaas For the call-gSNP run, it did INDEL realignment on the T/N pair together so that the samples were re-aligned together (BQSR happens at the read group level). So, you can use the realigned BAMs for call-sSNV without any issues. I would run the latest call-gSNP multi-sample mode to realign the 4 samples all together if you haven't.

> If it helps, I have 3 replicates for this T_radiosensitive sample.

If I recall, they are biological replicates, right (i.e. same experiment but cultured separately in the same environment -> different mutation distributions)?

**Reply by RoniHaas** [reply]
> .I would run the latest call-gSNP multi-sample mode to realign the 4 samples all together if you haven't.

By 4 samples, do you mean the 3 radiosensitive and one radioresistant?

>If I recall, they are biological replicates, right (i.e. same experiment but cultured separately in the same environment -> different mutation distributions)?

The definition of biological or technical is a bit tricky here I think. In terms of mutations, I tend to consider them as technical replicates.  They were taken from the same plate and then were grown separately before extraction.

**Reply by tnyamaguchi** [reply]
> > .I would run the latest call-gSNP multi-sample mode to realign the 4 samples all together if you haven't.
> 
> By 4 samples, do you mean the 3 radiosensitive and one radioresistant?

Yes. (I'm not sure if 2TB `/scratch` space is enough though..)

> > If I recall, they are biological replicates, right (i.e. same experiment but cultured separately in the same environment -> different mutation distributions)?
> 
> The definition of biological or technical is a bit tricky here I think. In terms of mutations, I tend to consider them as technical replicates. They were taken from the same plate and then were grown separately before extraction.

Yeah, so they were grown separately so you would expect different sets of somatic mutations from the 3 samples. The definition in this context is here Q1 - https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Dataset+registration+-+FAQ

@maotian06 is it possible to enable the `tumor-only` mode and `multi-sample` mode together in call-sSNV? It should be.

**Reply by maotian06** [reply]
@RoniHaas One concern is that you will get many false-positive SNVs using the `tumor-only model`.
So I wondered if you are using the cell line data, is it possible to generate a panel of normal samples to a `normal.bam` and then run it...

**Reply by maotian06** [reply]
@tyamaguchi-ucla  I tested with Ardy's sample with 7 bams, it works, so I think the `multi-sample` would be enough for the Roni's case.

Yes! `Multi-sample` and `Tumor-only` can work together! @RoniHaas

**Reply by RoniHaas** [reply]
> @RoniHaas One concern is that you will get many false-positive SNVs using the `tumor-only model`. So I wondered if you are using the cell line data, is it possible to generate a panel of normal samples to a `normal.bam` and then run it...

Since these are cell lines, I am not sure that creating a panel of N samples from some patient samples that we have in the lab is the best solution. I also only need an evaluation of the number of sSNVs in these samples. I am not going to use the positions probably. Maybe in my case single-tumor mode is good enough?
